internal/controller/pod/inithandler.go

```diff
@@ -291,3 +304,40 @@ func (c *Controller) labelPostgresPodAndDeployment(newpod *apiv1.Pod) {
 		return
 	}
 }
+
+// waitForPrimaryDeployment checks to see that a primary deployment is
+// available. It does not check readiness, only that the deployment exists. This
+// used before scaling to ensure scaling does not fail silently
+func (c *Controller) waitForPrimaryDeployment(cluster *crv1.Pgcluster) error {
+	ctx := context.TODO()
+	primaryDeploymentName := cluster.Annotations[config.ANNOTATION_CURRENT_PRIMARY]
+	options := metav1.ListOptions{
+		LabelSelector: fields.AndSelectors(
+			fields.OneTermEqualSelector(config.LABEL_PG_CLUSTER, cluster.Name),
+			fields.OneTermEqualSelector(config.LABEL_PG_DATABASE, config.LABEL_TRUE),
+			fields.OneTermEqualSelector(config.LABEL_DEPLOYMENT_NAME, primaryDeploymentName),
+		).String(),
+	}
+
+	// as the poll will initially wait 5 seconds, do a quick check to see if the
+	// deployment already exists, as most cases will fall into this
+	if d, err := c.Client.AppsV1().Deployments(cluster.Namespace).List(ctx, options); err == nil && len(d.Items) > 0 {
+		return nil
+	}
```

Comment:
:memo: [`wait.PollImmediate()`](https://pkg.go.dev/k8s.io/apimachinery@v0.20.1/pkg/util/wait#PollImmediate) will perform the check then wait.

Reply:
@cbandy thanks, i'll update to use that